### ISSUE TITLE
menu button

### ISSUE THREAD

**Issue by mojtaba82** [antroids/application-title-bar#13]
Great job. 
If you add Global Menu either as a single button or fully integrated with the title of window, it would be great. Similar to what Unity used to do.

**Comment by maverick74** [comment]
I don't wish to be the bearer of bad news but: https://github.com/antroids/application-title-bar/issues/10#issuecomment-1992273205

**Comment by antroids** [comment]
Yes, I don't see much sense in reimplementing the same functionality that Global Menu already has.
I can see that it works well and maintained as part of [Plasma Workspace](https://invent.kde.org/plasma/plasma-workspace/-/tree/master/applets/appmenu) project.
The Global menu has only private QML API without any guarantees that it will not be changed or even removed.
The other way would be copy-paste CPP plugin and include it to the widget, but I don't like this idea.

Are there any issues with using the both widgets on one panel?

**Comment by maverick74** [comment]
> Are there any issues with using the both widgets on one panel?

If you mean Application-title-bar  & Global menu, i had no problem with using them (when i tried both...).

in fact, AFAIK you can even use multiple instances of the same widget in the same panel (i used this in plasma 5 a few years ago where there was a [widget](https://github.com/kotelnik/plasma-applet-active-window-control) that had Window Control buttons and Global Menu AIO)

**Comment by antroids** [comment]
> in fact, AFAIK you can even use multiple instances of the same widget in the same panel (i used this in plasma 5 a few years ago when there was a [widget](https://github.com/kotelnik/plasma-applet-active-window-control) that had Window Control buttons and Global Menu AIO)

Yes, the widget could also be used as invisible spacer: #14

**Comment by maverick74** [comment]
I don't use any of the widgets ATM because of that global menu effect i referenced when i switch between supported apps and unsupported ones...

It has a terrible effect and distracts me, keeping me from using it. Since i would need both to accomplish the effect i wanted, i don't use any (but i still keep an eye on application-title-bar because if GM gets [that](application-title-bar ) fixed i'll be using both)...

**Comment by mojtaba82** [comment]
> Yes, I don't see much sense in reimplementing the same functionality that Global Menu already has. I can see that it works well and maintained as part of [Plasma Workspace](https://invent.kde.org/plasma/plasma-workspace/-/tree/master/applets/appmenu) project. The Global menu has only private QML API without any guarantees that it will not be changed or even removed. The other way would be copy-paste CPP plugin and include it to the widget, but I don't like this idea.
> 
> Are there any issues with using the both widgets on one panel?

I'm going to put the menu button between the app control buttons and the app title.

**Comment by maverick74** [comment]
> I'm going to put the menu button between the app control buttons and the app title.

I understand it's not perfect but, as a workaround, you can use 2 application-title-bar! One before and one after the global menu!

You'll still suffer the same problem I have, however!

(but yes, having an aio widget/having a global menu in application-title-bar would be great!)

**Comment by maverick74** [comment]
An update regarding the global menu: the problem i referenced is going to get fixed (probably) in kde 6.0.3 due to be released next week.

**Comment by KaKi87** [comment]
> Are there any issues with using the both widgets on one panel?

Hi,

It's not really an issue, but a UX advantage : when window controls and global menu are on the same panel like on Unity, the title is visible in the default state and the global menu replaces it on hover, allowing both to enjoy all the space, and specifically for the global menu, it doesn't have to either move horizontally or make empty space appear between apps.

Thanks

**Comment by MajorTomDE** [comment]
Maybe this fits to my proposal: https://github.com/antroids/application-title-bar/issues/31

To integrate the global menu into into a classic „left click“ menu on the title together with e.g. close/exit Function (because sometimes close/exit is missing for some applications)

**Comment by amorozov** [comment]
> Are there any issues with using the both widgets on one panel?

Actually, the only issue is the limited horizontal space. Take a look at the images:

![Image](https://github.com/user-attachments/assets/3925d336-ec3b-4589-883d-2f3d42b91554)

This is a normal, "acceptable" state.

![Image](https://github.com/user-attachments/assets/4f748067-4ba4-4fa8-8e8e-d5fe66213cb9)

and now the title + menu have pushed out to the right the rest of the panel widgets.

In general, the basic idea behind Unity behavior is that one doesn't need buttons + title and menu **at the same time**. If persons need to get an idea what application/documents they are working on, they look at the title. If persons need to perform some action available through the menu, they move their mouse/pointer over the title area and the title gets replaced with the menu. Thus we save a lot of horizontal space in the panel.

As alternative to a "big integrated widget" one could imagine two (or even three as my current installation has) widgets, each reacting on mouse hover events: buttons + title get hidden, while menu gets drawn. While this solution is more flexible than the "big widget", it seems that the visually unattractive flickering is inevitable.

**Comment by felagund** [comment]
> In general, the basic idea behind Unity behavior is that one doesn't need buttons + title and menu **at the same time**.

I actually never liked this behaviour, especially on larger screens. Quite often there is enough space for both, like for example on a 36 inch monitor. But I agree this is functionality some might  like and ideally would be possible to configure it.

In general, I would see three advantages to including the global menu in this widget:
1) Less configuration for the user. My preffered setup is mimicking Unity, so buttons <> menu <> window title. With three screens, that is nine widgets I need to configure. If it was one widget, I would configure it just once.
2) Uniformity of design. I set up the widget to only show for maximized windows (but also when they are not in focus, ie. focu has an unmaximized window dispalyed on top of them). Plasma's global manu has nothing like this capability, unfortunately. I would also like to be able to drag the maximized window by initiating the drag on the menu - now also impossible.

3) Being able to prioritize what is shown in case of not enough space. This would addres @amorozov's concern, where one could set his behaviour, or some sort of elision personally I would prefer the menu to always be shown and the title elided in the middle to fill the remaining space.

**Comment by KaKi87** [comment]
> I actually never liked this behaviour, especially on larger screens

Well, yeah, this UI was specifically designed with small screens in mind.

[I used Unity](https://old.reddit.com/r/UsabilityPorn/comments/12tlydg/unity_v77_in_ubuntu_2304_on_14_thinkpad_x260/) and [now KDE Plasma](https://old.reddit.com/r/UsabilityPorn/comments/1d7cw66/plasma_kde_neon_6_on_123_microsoft_surface_pro_5/) on my 12.3" Microsoft Surface Pro, but I rather use COSMIC from Pop OS on my triple monitor desktop setup, because one thing can't make sense everywhere everytime.
